presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -397,6 +397,31 @@ public synchronized void updatePartitionStatistics(String databaseName, String t
         return tables;
     }
 
+    @Override
+    public synchronized List<String> getTablesMatchingParameter(String databaseName, String parameterKey, String parameterValue)
+    {
+        requireNonNull(databaseName, "databaseName is null");
+        requireNonNull(parameterKey, "parameterKey is null");
+        requireNonNull(parameterValue, "parameterValue is null");
+
+        Optional<Database> database = getDatabase(databaseName);
+        if (!database.isPresent()) {
+            return ImmutableList.of();
+        }
+
+        Path databaseMetadataDirectory = getDatabaseMetadataDirectory(databaseName);
+        List<String> tables = getChildSchemaDirectories(databaseMetadataDirectory).stream()
+                .map(Path::getName)
+                .map(tableName -> getTable(databaseName, tableName))
```

Comment:
I think this should be 
```.map(tableName -> getTable(databaseName, tableName).orElseThrow(...)```
You know the database exists at this point

Reply:
I suppose the tables can be invalid here. The `tableName` is gotten by listing the directory but it may not be a valid one that has table metadata.

Reply:
If you look at `io.prestosql.plugin.hive.metastore.file.FileHiveMetastore#getAllTables` they don't call `getTable(databaseName, tableName)`

Reply:
Yes. But `getAllTables` doesn't involve getting table metadata from a file. I mainly referenced `getAllViews` for this method.